### DISCUSSION TITLE
RSC: server components should get getServerSideProps context as props

### DISCUSSION THREAD

**Discussion by davidpfahler** [vercel/next.js#35851] [Ideas]
### Describe the feature you'd like to request

In order so unlock the potential of RSC, every server components should be rendered with the `context` supplied to `getServerSideProps`, enabling all the server-side business logic at the component (not just page) level. The only part of `context` that wouldn't work seems to be `res`.

### Describe the solution you'd like

The following dream code illustrates how this would be super useful.

```typescript
// src/components/Sidebar.tsx

export const Sidebar = ({ req  }) => {
  if (req.cookies?.token != null) {
    const user = db.user.find(req.cookies.token);
    return <div>Welcome {user.name}!</div>
  }
  return <SignInButton />
}
```

### Describe alternatives you've considered

The only alternative right now seems to be to grab values off the `context` inside `getServerSideProps` and passing them down to components. This needs to be done at the page level and thus negates a lot of the benefits of RSC, namely to have the full power of the server at the component level.

**Comment by JimLiu** [comment]
Totally agree, at least we should be able to get `req`, `res` from `_document`!

**Comment by 0-don** [comment]
yes this is really needed
